text/trace/0226-sampling-random-traceids.md

```diff
@@ -0,0 +1,185 @@
+TraceID randomness.
+
+This proposes to extend that specification with support for 56-bit
+precision sampling probability.  This is seen as particularly
+important for implementation of probabilistic tail samplers (e.g., in
+the OpenTelemetry Collector) as explained below.
+
+This OTEP makes use of the [draft-standard W3C tracecontext `random`
+flag](https://w3c.github.io/trace-context/#random-trace-id-flag),
+which is an indicator that 7 bytes of true randomness are available
+for probability sampler decisions.
+
+## Explanation
+
+The existing, experimental TraceState probability sampling
+specification relies on two variables known as **r-value** and
+**p-value**.  The r-value carries the source of randomness and the
+p-value carries the effective sampling probability.
+
+Given this specification, a ConsistentProbabilitySampler can be
+applied as a head sampler for non-power-of-two sampling probabilities
+using interpolation.  For example, a neffective sampling probability
+of 1-in-3 can be achieved by alternating between 25% and 50% sampling.
+However, interpolation only works for trace roots, otherwise
+"consistent" sampling can only be achieved at the next smaller power
+of two.  In the example, sampling at 1-in-3 using interpolation means
+traces are only guaranteed **consistent** at 25% and smaller sampling
+probabilities.
+
+The major downside of the r-value, p-value approach is that r-value
+must be encoded even for unsampled contexts.  Ideally, building
+Span-to-Metrics pipelines should be low overhead which means not
+adding additional data to unsampled contexts.
+
+This proposal avoids r-value by using 7 bytes of intrinsic randomness
+in the TraceID, the ones (draft-) specified [in the W3C tracecontext
+`random` flag](https://w3c.github.io/trace-context/#random-trace-id-flag).
+Since this Sampler is expected to behave consistently with or without
+the `random` flag, we assumes the bits are random and do not actually
+check the W3C random flag.
+
+This document propose extending the existing p-value, r-value
+mechanism with support for a new indicator for non-power-of-two
+probability sampling known as "t-value", where "t" is chosen because
+it signifies a threshold.  If widely adopted, the tracestate r-value
+can be deprecated, as it is not needed when randomness is provided in
+the TraceID.
+
+As proposed, t-value and p-value are mutually exclusive; p-value
+remains the preferred encoding for probability sampling when a
+power-of-two sampling probability is used.  P-value also remains the
+specified way to encode zero adjusted count (i.e., p=63).
+
+### T-Value encoding
+
+Since we have 7 bytes, or 56 bits of randomness available, there are
+2^56 non-zero sampling probabilities that can be encoded.  These
+probabilities can be expressed as a 56-bit number in the range [0,
+0xffffffffffffff], where 0 corresponds with sampling 1 span out of
+2^56 and 0xffffffffffffff corresponds with sampling 100% of spans.
```

Comment:
I would prefer to define the t-value as the number of dropped spans out of 2^56. In this way the value 0 would correspond to 100% sampling and 0xffffffffffffffff would correspond to 1 out of 2^56. This would have the benefit that the hex-encoding (as described below) of probabilites which are powers of 2 would be much more efficient. For example, 100% would be "t:0", 50% would be "t:8", "25%" would be "t:c", "12.5%" would be "t:e", 6.25% would be "t:f0", and so on.

Another advantage is that with this definition the t-value increases with decreasing sampling probability, which is similar to the p-value.